src/utils/helper-functions/guards.ts

```diff
@@ -54,3 +55,21 @@ export function isIdentifierOrMemberExpression(
 export function isTypeReference(type: ts.Type): type is ts.TypeReference {
   return type.hasOwnProperty('target')
 }
+
+export function isCallExpressionWith(
```

Comment:
Not sure about the name + the place to put it.

Reply:
Either here or in utils.
Let's leave it here, if we need it somewhere else we can move it